sources/packages/api/src/route-controllers/assessment/assessment.system-access.controller.ts

```diff
@@ -0,0 +1,309 @@
+   * assessment.
+   * @param assessmentId assessment id to be updated.
+   * @param payload contains the workflowId to be updated.
+   */
+  @Patch(":id/workflow-id")
+  @ApiResponse({
+    status: HttpStatus.OK,
+    description: "Workflow id updated.",
+  })
+  @ApiResponse({
+    status: HttpStatus.UNPROCESSABLE_ENTITY,
+    description:
+      "Not able to update the assessment workflowId with provided data.",
+  })
+  async updateAssessmentWorkflowId(
+    @Param("id") assessmentId: number,
+    @Body() payload: UpdateAssessmentWorkflowIdDTO,
+  ): Promise<void> {
+    const updateResult = await this.assessmentService.updateWorkflowId(
+      assessmentId,
+      payload.workflowId,
+    );
+
+    // Checks if some record was updated.
+    // If affected is zero it means that the update was not successful.
+    if (updateResult.affected === 0) {
+      throw new UnprocessableEntityException(
+        "Not able to update the assessment workflowId either because the id was not found or the workflow id is already present.",
+      );
+    }
+  }
+
+  /**
+   * Updates the assessment data resulted from the
+   * assessment workflow process.
+   * @param assessmentId assessment to be updated.
+   * @param payload data to be persisted.
+   */
+  @Patch(":assessmentId/assessment-data")
+  @ApiResponse({
+    status: HttpStatus.OK,
+    description: "Assessment data saved.",
+  })
+  @ApiResponse({
+    status: HttpStatus.NOT_FOUND,
+    description: "Assessment id was not found.",
+  })
+  async updateAssessmentData(
+    @Param("assessmentId") assessmentId: number,
+    @Body() payload: UpdateAssessmentDataDTO,
+  ): Promise<void> {
+    const updateResult = await this.assessmentService.updateAssessmentData(
+      assessmentId,
+      payload.data,
+    );
+
+    if (updateResult.affected === 0) {
+      // No rows updated because the assessment id was not found.
+      throw new NotFoundException(
+        `Assessment id ${assessmentId} was not found.`,
+      );
+    }
+  }
+
+  /**
+   * Create the disbursements for an assessment/reassessment.
+   * Creates the disbursements and values altogether.
+   * @param assessmentId application id to associate the disbursements.
+   * @param payload array of disbursements and values to be created.
+   * @returns created disbursements ids.
+   */
+  @Post(":assessmentId/disbursements")
+  @ApiResponse({
+    status: HttpStatus.CREATED,
+    description: "Disbursements and disbursements values saved.",
+  })
+  @ApiResponse({
+    status: HttpStatus.NOT_FOUND,
+    description: "Assessment id was not found.",
+  })
+  @ApiResponse({
+    status: HttpStatus.UNPROCESSABLE_ENTITY,
+    description:
+      "The disbursement information is already present or either the assessment or the application is not in the correct state.",
+  })
+  async createDisbursement(
+    @Param("assessmentId") assessmentId: number,
+    @Body() payload: CreateDisbursementsDTO,
+  ): Promise<number[]> {
```

Comment:
Why are we returning the ids of the created disbursements. Asking for my understanding.

Reply:
So far when we are doing a POST, we are returning the id/ids of the created resources.
We started this a long time ago to try to make it a bit more "rest" in the sense that POST usually returns the 201 status and indicates in some way the resource created, in the location header usually.